arch/cpu/cc26x0-cc13x0/dev/gpio-hal-arch.c

```diff
@@ -43,18 +43,33 @@
 
 #include <stdint.h>
 /*---------------------------------------------------------------------------*/
-#define CONFIG_MASK (IOC_IOPULL_M | IOC_INT_M | IOC_IOMODE_OPEN_SRC_INV)
+#define CONFIG_MASK (IOC_IOPULL_M | IOC_INT_M | \
+                     IOC_IOMODE_OPEN_SRC_INV | IOC_HYST_ENABLE)
 /*---------------------------------------------------------------------------*/
 void
 gpio_hal_arch_no_port_pin_cfg_set(gpio_hal_pin_t pin, gpio_hal_pin_cfg_t cfg)
 {
-  uint32_t config;
-  gpio_hal_pin_cfg_t tmp;
+  /* Get current settings */
+  uint32_t config = ti_lib_ioc_port_configure_get(pin);
+
+  /* Check if we are enabling interrupts now */
+  bool enabling_interrupts = false;
```

Comment:
Writing this as
`bool enabling_interrupts = (config & IOC_INT_ENABLE) == 0 && (cfg & GPIO_HAL_PIN_CFG_INT_ENABLE);`
gives a single update of `enabling_interrupts`.

(The `config |= ...` below could also be rewritten with a ?: in the RHS, but that's not the style of the rest of the file.)

Reply:
:+1: Thanks for the review

> (The config |= ... below could also be rewritten with a ?: in the RHS, but that's not the style of the rest of the file.)

Yea, a lot of refactoring and styling opportunities in that file but I'll leave that for another PR